futurex_openedx_extensions/helpers/tenants.py

```diff
@@ -314,3 +316,46 @@ def get_tenants_sites(tenant_ids: List[int]) -> List[str]:
 def get_nafath_sites() -> List:
     """Get all nafath sites"""
     return get_all_tenants_info()['special_info']['nafath_sites']
+
+
+def generate_tenant_config(name: str) -> dict:
+    """
+    Generate a tenant configuration by copying the default config and replacing placeholder PARAM.
+
+    :param name: Name to replace 'PARAM' in the default configuration
+    :type name: str
+    :return: Updated tenant configuration with placeholders replaced
+    :rtype: dict
+    """
+    try:
+        default_config = TenantConfig.objects.get(route__domain=settings.FX_DEFAULT_TENANT_SITE)
+        config_lms_dict = default_config.lms_configs
+        updated_config = json.loads(json.dumps(config_lms_dict).replace('{{platform_name}}', name))
```

Comment:
`sub_domain` and `platform_name` are two different things
```suggestion
        updated_config = json.loads(json.dumps(config_lms_dict).replace('{{platform_name}}', platform_name))
        # replace {{sub_domain}}
        # replace {{course_org_filter}}
```